src/core/server/saved_objects/import/utils.ts

```diff
@@ -183,3 +183,38 @@ const parseJSONSpec = (spec: string) => {
 
   return undefined;
 };
+
+export function findReferenceDataSourceForObject(
+  savedObject: SavedObject,
+  savedObjects: Map<string, SavedObject>,
+  visited = new Set<string>()
+): SavedObjectReference | null {
```

Comment:
What if there are multiple data sources for a object? Like a dashboard.

Reply:
Sure, I am currently modifying the code here. After research, it seems that we can remove the `findReferenceDataSourceForObject` method and, all reference data sources of the copied object can be obtained from the `nestedDependencies`. Even if some data sources are removed, we can still get the ids in `nestedDependencies.missingRef`.

Reply:
updated